### PR TITLE
Add debug tips section to libcudf developer guide

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Adds a debugging tips section to the developer guide.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by ttnghia** [comment]
Do you think this is useful to add into the dev guide? This: https://nvidia.slack.com/archives/CDTANRCTT/p1679511691803229

I wanted to have a PR for it but never had time.

File: cpp/doxygen/developer_guide/DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -1384,3 +1384,24 @@ cuIO is a component of libcudf that provides GPU-accelerated reading and writing
...
+   known to produce false positives.
+3. `cudf::test::print()`\
+   The `print()` utility can be called within a gtest to output the data in a `cudf::column_view`.
+   More information is available in the [Testing Guide](TESTING.md#printing-and-accessing-column-data)
+4. GCC Address Sanitizer\
+   The GCC ASAN can also be used by adding the `-fsanitize=address` compiler flag.
+   There is a compatibility issue with the CUDA runtime that can be worked around by setting
+   environment variable `ASAN_OPTIONS=protect_shadow_gap=0` before running the executable.
+   Note also the CUDA `compute-sanitizer` can also be used with GCC ASAN with the
+   environment variable `ASAN_OPTIONS=protect_shadow_gap=0,alloc_dealloc_mismatch=0`
```
Comment (ttnghia) [inline cpp/doxygen/developer_guide/DEVELOPER_GUIDE.md:1407]:
```suggestion
   Note that the CUDA `compute-sanitizer` can also be used with GCC ASAN with the
   environment variable `ASAN_OPTIONS=protect_shadow_gap=0,alloc_dealloc_mismatch=0`.
```

File: cpp/doxygen/developer_guide/TESTING.md [hunk 2]
```diff
@@ -455,10 +455,19 @@ Column comparison functions in the `cudf::test::detail` namespace should **NOT**
 
 ### Printing and accessing column data
 
-`include/cudf_test/column_utilities.hpp` defines various functions and overloads for printing
+The `<cudf_test/debug_utilities.hpp>` defines various functions and overloads for printing
```
Comment (wence-) [inline cpp/doxygen/developer_guide/TESTING.md:458]:
```suggestion
The `<cudf_test/debug_utilities.hpp>` header defines various functions and overloads for printing
```

**Review by wence- — APPROVED** [review]

File: cpp/doxygen/developer_guide/DEVELOPER_GUIDE.md [hunk 3]
```diff
@@ -1384,3 +1384,25 @@ cuIO is a component of libcudf that provides GPU-accelerated reading and writing
...
+
+Here are some tools that can help with debugging libcudf (besides printf of course):
+1. `cuda-gdb`\
+   Follow the instructions in the [Contributor to cuDF guide](../../../CONTRIBUTING.md#debugging-cudf) to build
+   and run libcudf with debug symbols.
+2. `compute-sanitizer`\
+   The [CUDA Compute Sanitizer](https://docs.nvidia.com/compute-sanitizer/ComputeSanitizer/index.html)
+   tool can be used to locate many CUDA reported errors by providing a call stack
+   close to where the error occurs even with a non-debug build. The sanitizer includes various
+   tools including `memcheck`, `racecheck`, and `initcheck`. The `racecheck` and `initcheck` have been
```
Comment (PointKernel) [inline cpp/doxygen/developer_guide/DEVELOPER_GUIDE.md:1398]: Not sure if we want to mention them all but there is also `synccheck`.

**Review by PointKernel — APPROVED** [review]

**Review by ttnghia — APPROVED** [review]

File: cpp/doxygen/developer_guide/TESTING.md [hunk 4]
```diff
@@ -455,10 +455,19 @@ Column comparison functions in the `cudf::test::detail` namespace should **NOT**
...
 columns (`print`), converting column data to string (`to_string`, `to_strings`), and copying data to
-the host (`to_host`).
-
+the host (`to_host`). For example, to print a `cudf::column_view` contents or `column_wrapper` instance
+to the console use the `cudf::test::print()`:
+```
```
Comment (bdice) [inline cpp/doxygen/developer_guide/TESTING.md:462]: Enable C++ syntax highlighting.
```suggestion
```cpp
```

**Review by bdice — APPROVED** [review]
Nice. Thanks, @davidwendt!

**Pull request merged**